[DOC-591][doc][ybm][cli] Docs for Database audit logging commands (#25483)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-cloud/managed-automation/managed-cli/managed-cli-reference/managed-cli-db-audit-logging.md

---
title: ybm CLI cluster db-audit-logging
headerTitle: ybm cluster db-audit-logging
linkTitle: cluster db-audit-logging
description: YugabyteDB Aeon CLI reference Cluster Database Audit Logging Resource.
headcontent: Manage cluster database audit logging
menu:
  preview_yugabyte-cloud:
    identifier: managed-cli-db-audit-logging
    parent: managed-cli-reference
    weight: 20
type: docs
---

Use the `cluster db-audit-logging` resource to perform operations on a YugabyteDB Aeon cluster, including the following:

- enable, disable, and update database audit logging.
- get information about database audit logging.

> For information on database audit logging settings, refer to [Database Audit Logging](../../../../cloud-monitor/logging-export/#ysql-audit-logging-settings).

Reviewer: ```suggestion
For information on database audit logging settings, refer to [Database Audit Logging](../../../../cloud-monitor/logging-export/#database-audit-logging).
```

---
title: ybm CLI cluster db-audit-logging
headerTitle: ybm cluster db-audit-logging
linkTitle: cluster db-audit-logging
description: YugabyteDB Aeon CLI reference Cluster Database Audit Logging Resource.
headcontent: Manage cluster database audit logging
menu:
  preview_yugabyte-cloud:
    identifier: managed-cli-db-audit-logging
    parent: managed-cli-reference
    weight: 20
type: docs
---

> 

Reviewer: Check links

Use the `cluster db-audit-logging` resource to perform operations on a YugabyteDB Aeon cluster, including the following:

- enable, disable, and update database audit logging.
- get information about database audit logging.

> 

Reviewer: ```suggestion

For information on database audit logging settings, refer to [Database Audit Logging](../../../../cloud-monitor/logging-export/#ysql-audit-logging-settings).

```

## Syntax

```text
Usage: ybm cluster db-audit-logging [command] [flags]
```

> ## Commands

Author: I looked at the CLI docs. They marked— --cluster-name as an inherited option. Do we still have to do it?

> ## Commands

Author: I had a quick convo, and will add --cluster-name to all options.

### enable

> Enable database audit logging for a cluster.

Author: Isn't this `audit` logging, not a query? @ddhodge

> Enable database audit logging for a cluster.

Reviewer: my mistake, copy/paster error

| Flag | Description |
| :--- | :--- |
| --integration-name | Required. Name of the integration. |

> | --ysql-config | Required. The YSQL config to set up database audit logging.<br>Provide key-value pairs as follows:<br>`log_catalog=<boolean>, log_level=<LOG_LEVEL>, log_client=<boolean>, log_parameter=<boolean>, log_relation=<boolean>, log_statement_once=<boolean>`.<br>Default is `[]`. |

Reviewer: ```suggestion
| --ysql-config | Required. The [YSQL audit logging settings](../../../cloud-monitor/logging-export/#ysql-audit-logging-settings), provided as key-value pairs.<br>Arguments:<ul><li>log_catalog</li><li>log_level</li><li>log_client</li><li>log_parameter</li><li>log_relation</li><li>log_statement_once</li></ul> |
```

Author: I understand the format, but the way they're passing isn't intuitive enough to understand how to; for example, these arguments have to pass under a list of key-value pair as -  `[log_catalog=<boolean>, log_level=<LOG_LEVEL>, log_client=<boolean>, log_parameter=<boolean>, log_relation=<boolean>, log_statement_once=<boolean>]`

Also, we don't understand the value of those arguments of what type, i.e., boolean.

If I read the suggested change, this is how I understand `--ysql-config log_level=<boolean> log_statement_once=<boolean>` Is this the correct way to pass the ysql-config? @Siddarth-Baldwa

Reviewer: I think the example, when added, will clarify how to pass the values; and the detailed explanation of the options is provided in the link

Reviewer 2: Here is an example of a command: 
``` bash
./ybm cluster db-audit-logging update  --integration-name=new  --ysql-config="log_catalog=true,log_client=true,log_level=NOTICE,log_relation=true,log_parameter=true,log_statement_once=true"  --statement_classes="READ,WRITE,ROLE"  --cluster-name=vibrant-antlion  --wait
```
Does this help?

Author: This helps

> | --statement_classes | Required. The YSQL config statement classes.<br>Please provide key value pairs as follows:<br>`statement_classes=READ,WRITE,MISC`. |

Reviewer: ```suggestion
| --statement_classes | Required. The YSQL statements to log, provided as key value pair.<br>Argument: statement_classes. Options are READ, WRITE, or MISC. |
```

Reviewer: There are more options in the DB audit logs docs page linked earlier, are some missing, or are some not supported?

Reviewer 2: @ddhodge all the statement classes mentioned in the docs are supported

### disable

Disable database audit logging for a cluster if enabled.

| Flag | Description |
| :--- | :--- |
| -f, --force | Optional. Bypass the prompt for non-interactive usage. |

### describe

Fetch detailed information about the audit logging configuration for a cluster.

> ### update

Reviewer: see comments for enable.
